types/testutil/events.go

```diff
@@ -25,7 +25,8 @@ func MatchEvent(expected proto.Message, emitted sdk.Event) error {
 // the proto.MessageName of the passed msg.
 func GetEvent(msg proto.Message, events []sdk.Event) (e sdk.Event, found bool) {
 	eventName := proto.MessageName(msg)
-	for _, e := range events {
+	for i := len(events) - 1; i >= 0; i-- {
+		e = events[i]
```

Comment:
I actually think we should get the first event here and update the test to insert into the table on any `Given` steps. The use of the server method for a `Given` step is outdated and the reason why it was failing. The `When` step is what we are testing and `Given` should set up the scenario for the `When`.

Reply:
Might also be good to add a comment to this method regarding getting the first event found.

Reply:
Another solution would be to reset the context, which is what I did in the bridge integration tests, but I still think we should get the first event and insert on `When`.

Reply:
the test step was updated to insert here - [fix: use insert](https://github.com/regen-network/regen-ledger/pull/1414/commits/5b06367b45af420b5f184ce3a225848f24ccd0f5)

I updated the `GetEvent` afterwards as it seemed more natural to search for events by most recent 🤷🏻 but no strong feeling either way